firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -1014,7 +1014,11 @@ public WatchChange decodeWatchChange(ListenResponse protoChange) {
         hardAssert(
             !version.equals(SnapshotVersion.NONE), "Got a document change without an update time");
         ObjectValue data = decodeFields(docChange.getDocument().getFieldsMap());
-        Document document = new Document(key, version, data, Document.DocumentState.SYNCED);
+        // The document will be serialized again before being written to local storage, memoize the
+        // encoded form to avoid encoding it again.
```

Comment:
Slight wording suggestion:

```
// The document may soon be re-serialized back to protos in order to store it 
// in local persistence.  Memoize the encoded form to avoid encoding it again.
```